src/components/gas/GasSpeedButton.js

```diff
@@ -323,19 +323,30 @@ const GasSpeedButton = ({
   }, [currentNetwork, speeds]);
 
   const menuConfig = useMemo(() => {
-    const menuOptions = speedOptions.map(gasOption => ({
-      actionKey: gasOption,
-      actionTitle: upperFirst(gasOption),
-      icon: {
-        iconType: 'ASSET',
-        iconValue: GAS_ICONS[gasOption],
-      },
-    }));
+    const menuOptions = speedOptions.map(gasOption => {
+      const totalGwei = add(
+        gasFeeParamsBySpeed[gasOption]?.maxFeePerGas?.gwei,
+        gasFeeParamsBySpeed[gasOption]?.maxPriorityFeePerGas?.gwei
+      );
+      const gweiDisplay =
+        currentNetwork === networkTypes.polygon
```

Comment:
1. what about arb and optimism?
2. for safety for any future changes, let's use `isEIP1559LegacyNetwork` from `handlers/web3`

Reply:
Arbitrum and Optimism only have 1 gas option so they don't get the context menu, so this is a separate condition than `isEIP1559LegacyNetwork`